src/test-support/FakeKuma.ts

```diff
@@ -242,15 +245,19 @@ export class KumaModule {
         'kuma.io/service': service ?? this.serviceName(),
         'kuma.io/protocol': this.protocol(),
         ...(zone && { 'kuma.io/zone': zone }),
+        ...additionalTags,
```

Comment:
I'm getting more and more concerned about the fact that our mocks are slowly becoming more complicated than the application itself.

It took me a while the other day to figure out why something wasn't showing that I thought should have been. We are also removing type safety/validation in areas where its sometimes important to and adding sometimes incorrect 'alarms' everywhere instead (the `true` vs `1` thing in the name of "making things simpler" from the other day)

I've mentioned before the usefulness of being able to see the API JSON structure in the single isolated mock file. I can just open it up and see what it is without having to go on a trip down the rabbit hole.

i.e. `fake.kuma.` helpers shouldn't really be used as an `import`.

Generally, they should be used for fake **values** not entire JSON structures. What is happening is very similar to these `<DoEverythingComponent />`s, they eventually get so complicated that bugs start to appear that get hard to find and fix and everything generally gets harder and harder to work with.

Its almost even more important here as we don't really want to start manually writing tests for our mock files.

Just so its said again, I get the duplication thing/issue and these ideally would be autogenerated, so we need to try and strike a balance until we get to autogeneration. If we never get to autogenerated mocks then we will soon end up in a situation where are mocks are as difficult to work with as our application used to be.

And yes I know, types! Just so its said, the autogeneration could be using the generated API data types to do that, but I want to avoid using the sometimes incorrect manually written data types i.e. the sooner we can autogenerate all this the better.

Finally, when amending mocks for testing purposes please think twice before creating multiple functions that are essentially `import` to import little bits of entire JSON structure. Ok sometimes maybe thats what we want, but sometimes its better to make 'value' type helpers and then compose so you can see the entire structure of the JSON in the single file.

---

In saying all of the above, there is a perfect opportunity here to make a `fake.kuma.tags()` helper function, which might help to explain when its a good opportunity to use a `fake.kuma.` helper, and when its not.

`tags` are arbitrary key/values, the property names don't really matter to us, its just a `Record<string, string>` that are mostly anything but also can have special kuma.io ones in. So yes its a JSON structure but its pretty much a 'value' seeing as we don't really care what that structure is.

It would be nice to have a `fake.kuma.tags()`

```js
tags: fake.kuma.tags(),
// or
tags: {
  ...fake.kuma.tags(),
  "kuma.io/zone": "zone"
}
// or something similar to that ^
```

wdyt think of the `fake.kuma.tags` idea?

---

Just to be clear this is non-blocking for this PR, just I keep seeing PRs with these mocks changes in, and I don't want to block any fixes/features etc but I do want to make sure our mocks don't get into the same state as the application used to be.

Reply:
Yeah, I get your concern. The dataplane mocks are more difficult to follow/assess at a glance nowadays. We use those utilities for 5 mocked endpoints currently (though we could consider dropping the /dataplanes/_overview mock as we’re not using this endpoint currently).

I’ll think some more about this. Using a tags utility is definitely worthwile.

Reply:
> (though we could consider dropping the /dataplanes/_overview mock as we’re not using this endpoint currently).

I'm pretty sure this would be a good idea also, I think theres only repetition because we are supporting endpoints we don't use.

👍